src/types.rs

```diff
@@ -627,24 +629,32 @@ enum AbsurdComparisonResult {
     InequalityImpossible,
 }
 
+enum Rel {
+    Lt,
+    Le,
+}
+
+// Put the expression in the form lhs < rhs or lhs <= rhs.
+fn normalize_comparison<'a>(op: BinOp_, lhs: &'a Expr, rhs: &'a Expr)
+                                -> Option<(Rel, &'a Expr, &'a Expr)> {
+    match op {
+        BiLt => Some((Rel::Lt, lhs, rhs)),
+        BiLe => Some((Rel::Le, lhs, rhs)),
+        BiGt => Some((Rel::Lt, rhs, lhs)),
+        BiGe => Some((Rel::Le, rhs, lhs)),
+        _ => None,
+    }
+}
```

Comment:
This looks like it could be useful for other lints, maybe you should move that to `utils`. Also you need 3 `/` for the doc and I like it to be nicely markdownified¹
```rust
/// Put the expression in the form `lhs < rhs` or `lhs <= rhs`.
```

1: I’m sure that word exists in some dictionary :smile: